File: Software_Basics/python_demos/guess_the_number.py
import random  # gives us tools for picking random numbers

secret = random.randint(1, 20)  # a <= secret <= b
tries = 0
guess = 0 

print("I'm thinking of a number between 1 and 20")

# Repeat until the user guesses the secret number.
while guess != secret:
    text = input("Take a guess: ")  
    guess = int(text)  
    
    tries = tries + 1  
    if guess < 1 or guess > 20:
        print("That number is out of range. Try again.")
    elif guess < secret:
        print("Too low, try again.")
    elif guess > secret:
        print("Too high, try again.")
    else:
        print("You got it in", tries, "tries!")

File: Software_Basics/Java_script/guess_the_number.js
import * as readline from "node:readline/promises";
import { stdin as input, stdout as output } from "node:process";

const rl = readline.createInterface({ input, output});

try {
    const secret = Math.floor(Math.random() * (20)) + 1;
    let tries = 0;
    let guess = 0;

    console.log("Im thing about a number between 1-20");

    while (guess !== secret) {
        const text = await rl.question("Take a guess:");
        guess = parseInt(text, 10);

        tries = tries + 1;
        if(guess < 1 || guess > 20 ) {
            console.log("That number is out of range, try again");
        }
        else if( guess > secret) {
            console.log(" Guess is too high, try again");
        }
        else if(guess < secret) {
            console.log(" guess is too low, try again");
        }
        else{
            console.log("you got in with", tries, "tries!");
        }
    }
}
finally{ 
    rl.close();}
